tests/autoyast/repos.pm

```diff
@@ -30,7 +30,11 @@ use utils;
 sub run {
     # sles12_minimal.xml profile does not install "ip"
     assert_script_run 'ip a || ifstatus all';
-    quit_packagekit;
+    if (!check_var('DESKTOP', 'textmode')) {
+        quit_packagekit;
+        # poo#87850 wait for packagekit related zypper process to exit
+        assert_script_run 'until ! pgrep zypper; do sleep 1; done';
+    }
```

Comment:
as you observed this only in special cases I would really suggest you treat it as a product bug, detect the error condition properly and record a soft-failure before applying the workaround

Reply:
Hi, Oliver, I just got the debug log with zypper process, https://openqa.nue.suse.com/tests/5322481#step/repos/17 it showed the zypper process is 'zypper -n purge-kernels'. As my understanding, this zypper process is called by init after migration and used to clean the old kernels. So this is not the bug of packagekit, just a normal zypper process and we just need wait it to quit then run any cmd related zypper is ok. I will create a new public function with the 'assert_script_run 'until ! pgrep zypper; do sleep 1; done';' and call it when we found such issue. WDYT?

Reply:
Right. That makes more sense. At best we can find a way that works without sleep+poll. If the tool `inotifywait` is available in the installation in which you run tests you could likely run:

```bash
inotifywait -q -r -e close_write /var/log/zypper.log
```

to wait for all background zypper processes to finish. Or maybe you can ask zypper maintainers if there is a better way as it's a common use case that a user wants to call zypper and that is already blocked.

Reply:
Ok, I will try this inotifywait cmd firstly. Thanks.

Reply:
It seems the inotifywait is not supported on SLES15SP3 or it need install package. https://openqa.nue.suse.com/tests/5322777#step/repos/7

Reply:
As offline talk with Oliver, restore to my initial code. Now will fix a same issue happened on patch_sle also.

Wait for verification log:
patch_sle:
http://openqa.nue.suse.com/tests/5326409#live

repos:
http://openqa.nue.suse.com/tests/5326410#live

Reply:
Verified.